src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -0,0 +1,190 @@
+    }
+
+    async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+        let formattedPayload: { [key: string]: unknown };
+
+        if (!payment || !payment.paymentData) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        const { paymentData, methodId } = payment;
+        const { shouldSaveInstrument, shouldSetAsDefaultInstrument } = isHostedInstrumentLike(paymentData) ? paymentData : { shouldSaveInstrument: false, shouldSetAsDefaultInstrument: false };
+
+        const { isStoreCreditApplied : useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
+
+        if (useStoreCredit) {
+            await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
+        }
+
+        try {
+            const paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(methodId);
+            const { clientToken } = paymentMethod;
+            if (!clientToken) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+            const elements = this._getStripeElements();
+
+            const { paymentIntent, error } = await this._getStripeJs().confirmPayment({
+                elements,
+                redirect: 'if_required',
+            });
+
+            if (error) {
+                if (error.type === 'card_error' || error.type === 'invalid_request_error' || error.type === 'validation_error') {
+                    throw new Error(error.message);
+                }
+                throw new RequestError();
+            }
+
+            const { id: token } = paymentIntent ??  { id: '' };
+
+            formattedPayload = {
+                credit_card_token: { token },
+                vault_payment_instrument: shouldSaveInstrument,
+                confirm: false,
+            };
+
+            await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+            const paymentPayload = this._buildPaymentPayload(methodId, formattedPayload, shouldSetAsDefaultInstrument);
+
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._unmountElement();
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _buildPaymentPayload(methodId: string, formattedPayload: { [key: string]: unknown }, shouldSetAsDefaultInstrument: boolean | undefined): Payment {
+        const paymentData = shouldSetAsDefaultInstrument
+            ? { formattedPayload, shouldSetAsDefaultInstrument }
```

Comment:
I think we dont need this ternary as if `shouldSetAsDefaultInstrument` is undefined then it wont be make a difference, so i think we dont need this function. or is something else i might be missing?

`const paymentData = { formattedPayload, shouldSetAsDefaultInstrument }`

Reply:
I made changes to this ♻️